encryption.tex

```diff
@@ -0,0 +1,202 @@
+Third, when the client is ready to access its data, it simply loads the metadata produced in the provisioning step and performs an HTTP POST in order to recover the encryption key. This process of encrypting data is the provisioning step.
+
+\vskip 2cm
+\begin{figure}
+  \includegraphics[width=14cm]{images/image3.png}
+  \centering
+  \caption{LUKS device interaction with Tang Server}
+\end{figure}
+
+\newpage
+\subsection{Tang Puppet Profile and Role}
+
+\begin{lstlisting}[language=Java]
+#Tang Profile
+class profile::core::tang() {
+  #Variables
+  $packages = ['tang']
+  #Add require packages
+  package { $packages:
+    ensure => 'present',
+  }
+  #Ensure tang service is running
+  service { 'tangd.socket':
+    ensure  => 'running',
+    require => Package[$packages],
+  }
+  #Create tangd.socket override directory
+  file { 'tangd.socket.d':
+    ensure => 'directory',
+    path   => '/etc/systemd/system',
+    mode   => '0755',
+  }
+  #Create override config file
+  file { 'override.conf':
+    ensure  => 'present',
+    path    => '/etc/systemd/system/tangd.socket.d',
+    mode    => '0644',
+    require => File['tangd.socket.d'],
+    content => @(OVERRIDE/L)
+      [Socket]
+      ListenStream=
+      ListenStream=7500
+      | OVERRIDE
+  }
+  ->exec { '/usr/bin/systemctl daemon-reload':
+    refreshonly => true,
+    notify      => Service['tangd.socket']
+  }
+}
+\end{lstlisting}
+
+Tang profile handles the installation of the tangd.socket service, and then modifies it so it listens on port 7500 for incomming connections from clevis-dracut.
+
+\newpage
+\section{Lab Testing - Proof of Concept}
+\subsection{Kickstart Modifications - Use of Encryption in Provisioning Template}
+Since the drive must me encrypted with LUKS early during the provisioning, new Kickstart Provisioning Template and Partition Tables had to be created at Foreman.
+
+\vskip 0.5cm
+\begin{lstlisting}[language=bash]
+#Encrypted VDA - Partition Table
+
+ignoredisk --only-use=${BOOT_DEV}
+zerombr
+clearpart --drives=${BOOT_DEV} --all --initlabel
+part /boot     --size=1024 --asprimary --ondrive=${BOOT_DEV}
+part /boot/efi --size=200  --asprimary --ondrive=${BOOT_DEV} --fstype=efi
+part pv.boot   --size=1 --grow  --encrypted --passphrase=temppass --ondisk=${BOOT_DEV}
+volgroup ${BOOT_VG} pv.boot
+logvol /               --vgname=${BOOT_VG} --size=1 --grow --name=root
+\end{lstlisting}
+
+"Encrypted VDA" initialize the System disk with two regular partitions - /boot and /boot/efi - and then a PV, a VG and a LV, been the LV encrypted through LUKS with a temporary password
+\begin{lstlisting}[language=bash]
```

Comment:
It isn't a problem but I think this is causing `bind` to be bold typeface in the PDF as it is a bash built-in.

Reply:
fixed